common/service/host/kcs.c

```diff
@@ -97,6 +99,34 @@ void kcs_read(void *arvg0, void *arvg1, void *arvg2)
 			}
 
 		} else { // default command for BMC, should add BIC firmware update, BMC reset, real time sensor read in future
+			if (pal_immediate_respond_from_KCS(req->netfn, req->cmd)) {
+				do { // break if malloc fail.
+					uint8_t *kcs_buff;
+					kcs_buff = malloc(KCS_BUFF_SIZE * sizeof(uint8_t));
+					if (kcs_buff == NULL) { // allocate fail, retry allocate
+						k_msleep(10);
+						kcs_buff = malloc(KCS_BUFF_SIZE * sizeof(uint8_t));
+						if (kcs_buff == NULL) {
+							printf("[%s] Failed to malloc for kcs_buff\n",
+							       __func__);
+							break;
+						}
+					}
```

Comment:
Do we expect an allocation to work the second time after sleeping 10ms?
Have we seen an allocation fail then succeed during testing?

If not we probably just want to print the error and break the first time.

Reply:
OK, will remove the second time memory allocation.